Initial approach

kinbiko opened the pull request:
I see you've been working hard in your own time, and really appreciate it.
I'm creating this PR so that I can give you some feedback.

javarb committed:
Modified README

javarb committed:
Added Requirements

javarb committed:
DESIGH added

javarb committed:
add iml files

javarb committed:
Updated Readme

Commit:
Added query design

Commit:
First code approximation

Commit:
First code approximation

Commit:
Added gitignore

kinbiko commented on src/DatabasicCommands.java:
```diff
@@ -0,0 +1,4 @@
+// https://stackoverflow.com/a/19894176
+enum DataBasicCommands{
+    insert, help
```
Enums in Java should be `CAPITAL_SNAKE_CASE`. Although lower case will work, it can look confusing when used in code.
If you want these to be lower case when printing them, consider creating a method that makes them lower case. The best way of doing that is overriding the `toString()` method, as mentioned in the [highest voted answer](https://stackoverflow.com/a/19894094/2422278) in the stackoverflow link you mentioned.

```java
public String toString() {
    return name().toLowerCase();
}
```

javarb replied:
This is ok now.

kinbiko commented on src/InputProcessor.java:
```diff
@@ -0,0 +1,84 @@
+* Bold text https://stackoverflow.com/a/29109958 - http://ascii-table.com/ansi-escape-sequences.php
+*/
+class InputProcessor {
+        void validateInput(String[] args) {
```
We almost always use 4 spaces for indentation in Java, or alternatively use tabs.
Although personally I don't mind 8 spaces (because they force you to write methods with fewer degrees of indentation), this really is unconventional in Java.

Whether you choose 4 spaces or tabs, be sure to be consistent with all your code.

javarb replied:
This is ok also.

kinbiko commented on src/InputProcessor.java:
```diff
@@ -0,0 +1,84 @@
+                                                // Then we check the validity of the path, this can be in a method
+
+                                        } else {
+                                                System.out.println("Error: \u001B[1mYou have to specify a correct path\u001B[0m");
```
I'm guessing the `\u001B[1m` bit changes the colour?
Pro tip: Try using `System.err` instead of `System.out` for errors.

kinbiko replied:
Okay, so I know now that this isn't for errors.
How about making a `makeBold` method so that you can put the weird `\u001B[1m` stuff in only one place?

```java
String makeBold(String s) {
    //Implementation here
}
```

javarb replied:
Hi!, yes this is a really good idea. I will do it.

javarb replied:
Done!

kinbiko commented on src/InputProcessor.java:
```diff
@@ -0,0 +1,84 @@
+
+                                                System.out.println("Path to JSON file args[1] = " + args[1]);
+
+                                                // Then we check the validity of the path, this can be in a method
```
Yes! Good intuition!

kinbiko commented on src/InputProcessor.java:
```diff
@@ -0,0 +1,84 @@
+                System.out.println("-------------------");
+                System.out.println("Available commands are:");
+                // To print all the enum values
+                System.out.println(java.util.Arrays.asList(DataBasicCommands.values())+"\n");  // https://stackoverflow.com/a/14413618
```
Clever!

javarb replied:
Thanks!

kinbiko commented on src/Main.java:
```diff
@@ -0,0 +1,33 @@
+        userInput = scanner.nextLine();*/
+
+        // Check if input isn't empty
+        InputProcessor inputProcessor = new InputProcessor();
```
I really like that you delegated to another class here!
For reasons that I'll get into later, it's a good idea to create a new object in `main()` and have your application somewhere other than `main`.

As a rule of thumb, I try and always make my `main` method be 1 line long. You can do this too, if you do: 

```java
new InputProcessor().validateInput(args)
```

Obviously, this won't be enough once we start writing more and more of the application. So in the future we would probably do something like:

```java
new Application().start(args)
```

Where `Application` is a class that handles validating input, delegating to insert and query methods, etc.

javarb replied:
Done, thanks!

kinbiko commented on README.md:
```diff
@@ -1,81 +1,88 @@
-  - Clean Code
-  - Clean Architecture
+1. Compatible only with *nix operating systems. 
+2. The software will run in CLI mode.
```
Pro tip: In markdown, it doesn't matter which number you put first, it will always show up correctly. I tend to always use `1.`. The reason for this, is so that if I ever need to add a bullet point between 4 and 5, I don't have to rename 5, 6, 7, 8, etc. So:

```
1. Compatible only with *nix operating systems. 
1. The software will run in CLI mode.
1. (...)
```

javarb replied:
Done! I understand and thanks for the tip. I have a question, about to add a bullet point between 4 and 5, you mean to change the number 4 or 5 for a bullet? Because just for insert a bullet between them I can do it using:

4. This
- blah
5. That 

Thanks!

kinbiko replied:
No no, I do mean a numbered list item, not an un-numbered bullet point :)
My bad.

javarb replied:
Ah yes, that would be a headache try to insert an numerate item there...

kinbiko reviewed (commented):
This is really good progress!

Commit:
Added file opening check

kinbiko commented:
Good stuff. Be sure to push your changes so I can merge them before continuing your work :)

javarb commented:
Hello Roger, I still have to remove that bunch of comments and put them inside notes. So I think I will uploading this soon. I must tell you by here when I do the push?

kinbiko commented:
Nope, no worries. I’ll see when you’ve pushed. Just wanted to make sure you hadn’t accidentally forgotten to push because it looked like you had addressed all the comments.

> On 13 Jun 2018, at 21:28, Javier Alexander Arboleda Rojas <notifications@github.com> wrote:
> 
> Hello Roger, I still have to remove that bunch of comments and put them inside notes. So I think I will uploading this soon. I must tell you by here when I do the push?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

javarb commented:
I understand, thanks for the clarification. Then I will push soon

El mié., 13 jun. 2018 a las 15:40, Roger Guldbrandsen (<
notifications@github.com>) escribió:

> Nope, no worries. I’ll see when you’ve pushed. Just wanted to make sure
> you hadn’t accidentally forgotten to push because it looked like you had
> addressed all the comments.
>
> > On 13 Jun 2018, at 21:28, Javier Alexander Arboleda Rojas <
> notifications@github.com> wrote:
> >
> > Hello Roger, I still have to remove that bunch of comments and put them
> inside notes. So I think I will uploading this soon. I must tell you by
> here when I do the push?
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub, or mute the thread.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/kinbiko/databasic/pull/1#issuecomment-397078888>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AKLmP9lG2cgI5mTLngpcIx1g10ohavgzks5t8XjKgaJpZM4UhUPY>
> .
>


-- 
JAVIER ALEXANDER ARBOLEDA ROJAS

javarb committed:
Applying suggestions to make code cleaner and shorter

javarb commented:
Hello Roger! I know this is not needed to say but the push is Done! Could you please check if there is something in my [notes](https://github.com/javarb/notes) I must to add or modify?

kinbiko commented:
I'm very happy with the way you've written your notes, and the way you've addressed my review comments! Keep it up!

javarb commented:
Thank you!
